### PR TITLE
Ontology guide update with new hygiene test descriptions

### PR DESCRIPTION
**Pull request by mereolog**
Signed-off-by: Pawel Garbacz pawel.garbacz@makolab.com

## Description

This pr updates ONTOLOGY_GUIDE.md with information about new hygiene tests.

Fixes: #1140

## Checklist:

- [X] I'm familiar with the [FIBO developer quide](../CONTRIBUTING.md#contributing-to-the-fibo-code). My contribution meets all the requirements described there.
- [X] My contribution follows the [principles of best practices for FIBO](../ONTOLOGY_GUIDE.md).
- [X] My changes have been reconciled with latest master and no merge conflicts remain.
- [X] This PR is related to exactly one issue. The issue is referenced by using a GitHub keyword such as "fixes", "closes", or "resolves".
- [X] Hygiene tests have been applied by a PR with "(WIP)" in title.
- [X] The issue has been tested locally using a reasoner (for ontology changes).

### CODE REVIEW

**Review by trypuz — DISMISSED** [review]
Approved

File: ONTOLOGY_GUIDE.md [hunk 1]
```diff
@@ -273,137 +273,75 @@ Because we believe that it is important to ensure at least some minimal level of
...
+1. **Circular Definitions** - No Fibo definition should be circular, i.e., its definiens cannot contain any resource defined by means of its definiendum. See [testHygiene1068.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1068.sparql)
+1. **Object Property Inverses** - No property may have more than one inverse. See [testHygiene1078.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1078.sparql)
+1. **Use of rdfs:comment** - rdfs:comment shouldn't be used for FIBO annotations. See [testHygiene1079.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1079.sparql)
+1. **Reference to owl:Thing** - We should not make explicit references to owl:Thing as these are obsolete. See [testHygiene0268.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene0268.sparql)
+1. **Synonyms as classes** - Use of owl:equivalenTo may indicate that different synonyms are modeled as separate classes, which should be avoided. See [testHygiene1103.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1103.sparql)
```
Comment (rivettp) [inline ONTOLOGY_GUIDE.md:282]: There is no owl:equivalentTo - it's owl:equivalentClass or owl:equivalentProperty. You could alternatively use the less formal "Use of OWL equivalences may indicate..."

File: ONTOLOGY_GUIDE.md [hunk 2]
```diff
@@ -273,137 +273,75 @@ Because we believe that it is important to ensure at least some minimal level of
...
+1. **Special Characters** - The set of characters allowed in literals is limited to the alphanumeric characters, punctuation marks, and diacritic characters used in the languages FIBO uses. See [testHygiene0114.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene0114.sparql)
+1. **Polysemous elements** - We should avoid using polysemous names, in particular no label can name more than one owl:Class or rdf:Property. See [testHygiene1067.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1067.sparql)
+1. **Circular Definitions** - No Fibo definition should be circular, i.e., its definiens cannot contain any resource defined by means of its definiendum. See [testHygiene1068.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1068.sparql)
+1. **Object Property Inverses** - No property may have more than one inverse. See [testHygiene1078.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1078.sparql)
```
Comment (rivettp) [inline ONTOLOGY_GUIDE.md:279]: you might want to add that it would make the inversed properties equivalent.

File: ONTOLOGY_GUIDE.md [hunk 3]
```diff
@@ -273,137 +273,75 @@ Because we believe that it is important to ensure at least some minimal level of
...
 1. **Labels and Definitions** - Every Class and Property defined in FIBO must have an rdfs:label and a skos:definition. See [testHygiene0004.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene0004.sparql)
 1. **Ontology Metadata** - Every Ontology defined in FIBO must have a rdfs:label, sm:copyright, dct:license, dct:abstract. See [testHygiene0005.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene0005.sparql)
+1. **Special Characters** - The set of characters allowed in literals is limited to the alphanumeric characters, punctuation marks, and diacritic characters used in the languages FIBO uses. See [testHygiene0114.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene0114.sparql)
+1. **Polysemous elements** - We should avoid using polysemous names, in particular no label can name more than one owl:Class or rdf:Property. See [testHygiene1067.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1067.sparql)
```
Comment (rivettp) [inline ONTOLOGY_GUIDE.md:277]: 2 issues here: a) I don't think many people will know what polysemy means; b) reflecting a) the hygiene test is not limited to polysemy but also catches homonymy - see https://en.wikipedia.org/wiki/Polysemy . Better IMO to talk in terms of "unique label rule" which is more comprehensible and reflects both the intent and logic

File: ONTOLOGY_GUIDE.md [hunk 4]
```diff
@@ -273,137 +273,75 @@ Because we believe that it is important to ensure at least some minimal level of
...
+1. **Circular Definitions** - No Fibo definition should be circular, i.e., its definiens cannot contain any resource defined by means of its definiendum. See [testHygiene1068.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1068.sparql)
```
Comment (rivettp) [inline ONTOLOGY_GUIDE.md:278]: I think use of "any resource defined by" is not really correct - we're just saying that a definition cannot contain a label (including synonyms) of the resource being defined.
Again, the use of Latin is unlikely to be understood.
BTW FIBO should be in caps

File: ONTOLOGY_GUIDE.md [hunk 5]
```diff
@@ -273,137 +273,75 @@ Because we believe that it is important to ensure at least some minimal level of
...
+1. **Object Property Inverses** - No property may have more than one inverse. See [testHygiene1078.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1078.sparql)
+1. **Use of rdfs:comment** - rdfs:comment shouldn't be used for FIBO annotations. See [testHygiene1079.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene1079.sparql)
+1. **Reference to owl:Thing** - We should not make explicit references to owl:Thing as these are obsolete. See [testHygiene0268.sparql](https://github.com/edmcouncil/fibo/blob/master/etc/testing/hygiene/testHygiene0268.sparql)
```
Comment (rivettp) [inline ONTOLOGY_GUIDE.md:281]: a) it's "redundant" not "obsolete"
b) there is one case where owl:Thing is needed which is for someValuesFrom restrictions.

**Review by rivettp — CHANGES_REQUESTED** [review]
A few clarifications and corrections

File: ONTOLOGY_GUIDE.md (hunk 3 above)
Comment (mereolog) [reply ONTOLOGY_GUIDE.md:277]: Agreed.

**Review by trypuz — APPROVED** [review]
Approved

**Review by rivettp — APPROVED** [review]

**Pull request merged**